packages/ra-tree-core/src/reducers/nodes.ts

```diff
@@ -0,0 +1,179 @@
+import { DELETE, DELETE_MANY, Identifier } from 'ra-core';
```

Comment:
The code of this file is very hard to read and understand. Maybe by introducing temporary constants, that would make the code clearer? Or maybe extract some of the processing into functions.